### ISSUE TITLE
Why is FirstName a required field?

### ISSUE THREAD

**Issue by elliot-sawyer** [silverstripe/silverstripe-framework#8803]
## Affected Version

4.3

## Description

I’m working on a site that only requires an email address and password to get started. I started out using memberprofiles to do a lot of the heavy lifting, and one feature of that module is choosing which fields to expose on your registration and profile pages using the CMS.  I found that when hiding the FirstName field, I was not able to continue with account signups because FirstName is a required field.

There are some changes I can pass in with extensions and via Config API to make sure that it works. After doing all of that I was left wondering, why is this field required at all? I would speculate that most sites nowadays do not require more than an email address or password to create a basic account. It would be nice if future versions of SilverStripe could make the FirstName field optional.

## Steps to Reproduce

1. Log into the CMS as an admin
2. Start adding a new Member via /admin/security.
3. Omit the FirstName field
4. Supply a valid email address and password.
5. Save, and observe the "FirstName is required" field.

Alternative, discovered via MemberProfiles
1. Install MemberProfiles
2. Edit the Account Page > Profile Fields to hide the FirstName field
3. Visit `http://yoursite/account` (or whatever your MemberProfilePage URL is) when not logged in.
4. Try to register and submit. Validation will fail.

**Comment by chillu** [comment]
Good point, I don't think it should be. This is done in [Member_Validator](https://github.com/silverstripe/silverstripe-framework/blob/4/src/Security/Member_Validator.php). It's technically an API change to remove that requirement though, certain member registration flows might rely on it. You could make a PR to disable this in 5.x, and add some docs which make the configuration option more obvious in 4.x?

**Comment by elliot-sawyer** [comment]
There was some discussion about it on the SS user's channel.

@robbieaverill shared a config doing it purely with the injector:
```
SilverStripe\Core\Injector\Injector:
  SilverStripe\Security\Member_Validator:
    calls:
      removeFirstName: [ removeRequiredField, [ FirstName ] ]
```

@edlinklater suggested copying the email address into FirstName as a quick fix. The only place I've found where it's used is when you try to log into the CMS without any permissions "You are logged in as . " with the "Log in as someone else" button. I think that string ought to be configurable, with a fallback to whatever the `unique_identifier_field` is for the authenticator.

**Comment by elliot-sawyer** [comment]
Happy to investigate a PR. What would the new expected behaviour be @chillu ?

**Comment by chillu** [comment]
Yep, check what in core relies on $FirstName - falling back to unique_identifier_field is a good call there. Case by case, not in `Member->getFirstName()`. Default behaviour in 5.x should be to only require Email. Maybe also check the 4.x docs around `unique_identifier_field` and see if that gotcha is mentioned in there?
